src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,286 @@
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch((err) => {
+				// eslint-disable-next-line no-console
+				console.error('error: ', err);
+			});
+	}
+
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${ES_DOMAIN}/edu-sharing/oauth2/token`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+
+			// eslint-disable-next-line max-len
+			body: `grant_type=${ES_GRANT_TYPE}&client_id=${
+				ES_CLIENT_ID
+			}&client_secret=${ES_OAUTH_SECRET}&username=${
+				ES_USER
+			}&password=${ES_PASSWORD}`,
+			timeout: REQUEST_TIMEOUT,
+		};
+		return request(oauthoptions).then((result) => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			// eslint-disable-next-line no-console
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	checkEnv() {
+		return (
+			ES_DOMAIN
+			&& ES_USER
+			&& ES_PASSWORD
+			&& ES_GRANT_TYPE
+			&& ES_OAUTH_SECRET
+			&& ES_CLIENT_ID
+		);
+	}
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async FIND(data) {
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = parseInt(data.query.$skip, 10) || 0;
+		const maxItems = parseInt(data.query.$limit, 10) || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = false;
+		const propertyFilter = '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || '';
+
+		if (!this.checkEnv()) {
+			return 'Update your env variables. See --> src/services/edusharing/envTemplate';
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const urlBase = `${ES_DOMAIN}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?`;
+		const url = urlBase
+			+ [
+				`contentType=${contentType}`,
+				`skipCount=${skipCount}`,
+				`maxItems=${maxItems}`,
+				`sortProperties=${sortProperties}`,
+				`sortAscending=${sortAscending}`,
+				`propertyFilter=${propertyFilter}`,
+			].join('&');
+
+		const options = {
+			method: 'POST',
+			// This will be changed later with a qs where sorting, filtering etc is present.
+			// eslint-disable-next-line max-len
+			url,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			body: JSON.stringify({
+				criterias: [
+					{ property: 'ngsearchword', values: [`${searchWord}`] },
+				],
+				facettes: ['cclom:general_keyword'],
+			}),
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const parsed = await requestRepeater(options);
+
+		// provided by client eg data.query.filterOptions
+		const filterOptions = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['filterOptions' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)